server/channels/app/content_flagging.go

```diff
@@ -603,6 +603,22 @@ func (a *App) PermanentDeleteFlaggedPost(rctx request.CTX, actionRequest *model.
 		return model.NewAppError("PermanentlyRemoveFlaggedPost", "app.content_flagging.permanently_delete.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
 
+	contentReviewBot, appErr := a.getContentReviewBot(rctx)
+	if appErr != nil {
+		return appErr
+	}
+
+	// If the post is not already deleted, delete it now.
+	// This handles the case when "Hide message from channel while it is being reviewed" setting is set to false when the post was flagged.
+	if flaggedPost.DeleteAt == 0 {
+		// DeletePost is called to care of WebSocket events, cache invalidation, search index removal,
+		// persistent notification removal and other cleanup tasks that need to happen on post deletion.
+		_, appErr = a.DeletePost(rctx, flaggedPost.Id, contentReviewBot.UserId)
```

Comment:
question: Where does the post gets permanently deleted? OR we just naturalizing the post by deleting contents? If so, I think the function name is misleading.

Reply:
Is this done? Anyway it's not blocking. I'll leave it to you if you want to change the name of the function later.